### ISSUE TITLE
WinJS.UI.Pages.define "ready" event firing multiple times, exponentially

### ISSUE THREAD

**Issue by appeality** [winjs/winjs#1200]
The weirdest thing is happening when I navigate to a particular page control, hit the back button, then navigate to the same page again.

2 issues:

1) the same page is added(duplicated) to WinJS.Navigation.history.backstack. This causes the same page to reload x number of times (depending on how many duplicate entries are in the backStack). I did this workaround (on the unload event) to see if this was the cause of the duplicate event firing:

WinJS.Navigation.history.backStack.splice(1, WinJS.Navigation.history.backStack.length - 1);

This fixed the navigation issue (I can now get directly back to the page originally navigated from), but the issue 2 is still present.

2) Each time I navigate to the page control, the ready event fires exponentially. For example, if I navigate to and from the page 4 times, on the 5th navigation, the ready event fires 4 times! Is there another location tracking the pages besides Navigation.history object?

**Comment by appeality** [comment]
Ok, so the actual issue was an app bar command's click event on the originating page being wired up multiple times. This is strange because I removed the handler before re-wiring, and no luck. For now I am using the jQuery.one() method, but its not ideal.

**Comment by AmazingJaze** [comment]
@appeality Do you think this is a problem with the WinJS.UI.AppBarCommand? If so I would like to fix it.
Can you provide a page/app which repros this behavior? You can share it using OneDrive as described [on our wiki](https://github.com/winjs/winjs/wiki/How-to-share-repros-of-issues-with-WinJS).

**Comment by appeality** [comment]
@AmazingJaze I struggled with this for a few hours. I narrowed the issue down to the method used to wire the click event of the WinJS.UI.AppBarCommand. Ultimately, I even had problems with the jQuery.one() method. 

However, your suggestion to use cmdSomeCommand.winControl.onclick = ("click", function(...)) did the trick. This seems to be the only reliable way to wire up the app bar commands and "magically" prevent the duplicate firing issue. I tried several different methods, and this is the only one that worked consistently. I will reproduce the issue and submit via the method above.
